ap_src/ap_app/utils/teams_utils.py

```diff
@@ -0,0 +1,44 @@
+# Third-party Libraries
+import hashlib
+import requests
+
+# Environment
+import environ
+
+env = environ.Env()
+environ.Env.read_env()
+
+
+def retrieve_calendar_data(user, sortValue):
+    data = None
+    r = None
+
+    try:
+        token = (str(user) + "AbsencePlanner").encode()
+        encryption = hashlib.sha256(token).hexdigest()
+        r = requests.get(env("TEAM_DATA_URL") + "api/user/teams/?format=json&username={}&sort={}".format(user.username, sortValue), headers={"TEAMS-TOKEN": encryption})
+    except:
+        print("API Failed to connect")
+    
+    if r is not None and r.status_code == 200:
+        data = r.json()
+    
+    return data
+
+def get_users_sharing_teams(current_user, user_model):
+    teams = retrieve_calendar_data(user_model, None)
+    if teams is None or teams == []: # The current_user is not in any teams
+        return {} # Avoid error from iterating through None type
+
+    users_sharing_teams = []
```

Comment:
Here we are setting 'user_sharing_teams' to an empty list. But later we convert it to a set. It's good practice to not switch the type of a variable. So you could define it as:
`users_sharing_teams = set()`

And then instead of `users_sharing_teams.append(username)` you could use a new variable here.

Reply:
This was a mistake with understanding the syntax

Reply:
Fixed with ccaa924